### PR TITLE
HPC-9742 Modify button component style

### PR DESCRIPTION
**Pull request by Onitoxan**
## Nature of PR

- [ ]  Bug-fix
- [ ]  Hot-fix
- [X]  Feature
- [ ]  Testing
- [ ]  Rewrite
- [ ]  CI
- [ ]  Package update

## Description

For more information read info in ticket HPC-9742. This PR includes code for modifying and adding new styles to our existing button for it to have more cohesion with how we are styling our components.

## How to test changes

Verify that the modified styles show up as expected.
## TODO:

No todo

### CODE REVIEW

**Review by unocha-hpc — APPROVED** [review]
Checks have passed and this pull request is ready for manual review

File: libs/hpc-ui/src/lib/components/button.tsx [hunk 1]
```diff
@@ -127,11 +130,22 @@ const StyledBaseButton = styled(BaseButton)`
   &:hover,
   &:focus,
   &.${CLS.ACTIVE} {
+    transition: ease-out 0.1s;
     text-decoration: none;
     background-color: ${(p) => p.theme.colors.pallete.gray.light};
     color: #fff;
   }
+  &.${COLOR_CLS.neutral_light} {
```
Comment (Pl217) [inline libs/hpc-ui/src/lib/components/button.tsx]: This looks little out of place. You have added this definition where empty line was, which makes git report as single line was removed, while entire PR adds new code.

There is a certain order established in `COLOR_CLS` and it goes `primary`, `primary_light`, `secondary`, `secondary_light`, `neutral`, `neutral_light`, so I'd respect this order when using the variables and place `neutral_light` after `secondary_light`.

Not very important, but contributes to better readability.

**Review by Pl217 — CHANGES_REQUESTED** [review]
One minor thing I'll address myself and then merge this

**Review by Pl217 — APPROVED** [review]

**Review by unocha-hpc — APPROVED** [review]
Checks have passed and this pull request is ready for manual review

**Pull request merged**